content/posts/windows-containers.md

```diff
@@ -0,0 +1,231 @@
+3. Create a new resource group (if needed)
+4. Create an AKS cluster
+    * You can use ```--nodepool-name``` with ```aks create cluster``` to name your control plane node pool i.e ```default```
+5. Add a Windows Server node pool 
+    * This will be a node pool for kubernetes pod Windows agents, we can name it ```--name winage```
+
+[Node pools](https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools) give us the possibility to extend our Kubernetes cluster with more types of machines depending on our use and budget (See available [options and default values](https://docs.microsoft.com/en-us/cli/azure/ext/aks-preview/aks/nodepool?view=azure-cli-latest#ext-aks-preview-az-aks-nodepool-add)) for AKS. As an example, we are going to add two more identical pools (one for masters and one for Linux agents) but you can pick different machine sizes and node counts depending on your need (just make sure that the VMs used for Jenkins masters support Premium Storage as Jenkins requires high IOPS for better performance):
+    
+* Linux Jenkins master pool example:
+
+    ```    
+    az aks nodepool add \ 
+    --resource-group myResourceGroup \
+    --cluster-name eastUSAKS \
+    --os-type Linux \
+    --name masters \
+    --node-count 1 \
+    --kubernetes-version 1.14.6 \
+    --node-vm-size Standard_DS2_v2
+    ```
+
+* Linux Jenkins agent pool example:
+
+    ```
+    az aks nodepool add \                                                                 --resource-group myResourceGroup \
+    --cluster-name eastUSAKS \
+    --os-type Linux \
+    --name linage \
+    --node-count 1 \
+    --kubernetes-version 1.14.6 \
+    --node-vm-size Standard_DS2_v2
+    ```
+    Once the pools are created, you can see them in the Azure portal:
+
+    [![](/img/windows-containers/AKS-node-pools.png)](/img/windows-containers/AKS-node-pools.png)
+
+
+## Jenkins installation using Helm
+
+[Helm](https://helm.sh/) is the Kubernetes Package Manager and we can use it to install Jenkins using a chart([Jenkins chart](https://github.com/helm/charts/tree/master/stable/jenkins)). If you haven't installed Helm before, you can follow [these instructions](https://docs.microsoft.com/en-us/azure/aks/kubernetes-helm) to install it. Using a [nodeSelector](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/) in the values file (```values.yaml```) used by the chart will allow us to specify in which nodepool Jenkins will be installed (you can find ```master.nodeSelector``` option in the Jenkins chart link). For simplicity, we are only going to configure the ```values.yaml``` file so that it deploys Jenkins using such ```nodeSelector``` option but the file can include a lot more options.  In this case, we need to make sure that Jenkins runs in the nodepool named ```masters``` (AKS assigns the nodepool name as the value of the ```agentpool``` tag, more on this in the next section)
```

Comment:
The first sentence here might be cleaner ending like:
"...to install Jenkins using the [official chart](https://github.com/helm/charts/tree/master/stable/jenkins)."

The last sentence is missing ending punctuation.